src/lib/libstrings.js

```diff
@@ -162,6 +176,28 @@ addToLibrary({
     if (!(maxBytesToWrite > 0))
       return 0;
 
+#if TEXTENCODER == 2
+    // Always use TextEncoder when TEXTENCODER == 2
+    var encoded = UTF8Encoder.encode(str);
+    var bytesToWrite = Math.min(encoded.length, maxBytesToWrite - 1); // -1 for null terminator
+    for (var i = 0; i < bytesToWrite; ++i) {
+      heap[outIdx + i] = encoded[i];
+    }
+    heap[outIdx + bytesToWrite] = 0;
+    return bytesToWrite;
+#else
+    // When using conditional TextEncoder, use it for longer strings if available
+    if (UTF8Encoder) {
```

Comment:
The comment sets its only being used to longer strings but the code doesn't do that.   Do we want to do it?

Reply:
done